mono/utils/mono-dl-windows.c

```diff
@@ -157,7 +157,21 @@ mono_dl_lookup_symbol (MonoDl *module, const char *symbol_name)
 
 	/* get the symbol directly from the specified module */
 	if (!module->main_module)
+	{
```

Comment:
Should we only do this for none main module or should we add same capabilities to lookup by ordinal in main module as well?

Reply:
Well, "__Internal" is a Mono-only feature, right? So doing that wouldn't add compatibility with .NET Framework. I'm not sure there's any other reason to add this. If we did, we'd only want to search the main exe and not every loaded module, because collisions are likely.

Reply:
It will look in the main binary (the .exe) in case main_module is true, so its more a mono feature and not something you use in C# dllimport's, but since this is generic library code used from within runtime, I don't see why we should restrict this use case compared to cases when you just use a symbol name.

Reply:
What are the possible use cases for this? Either you want to use an export from the mono runtime itself (in which case the export should have a name anyway, using it this way would be less portable, and the ordinal is probably inconsistent as exports change), or you are embedding Mono in a mixed-mode or native exe that has ordinal exports. I guess I could see an argument for this in the unlikely case that the runtime is embedded using a generic plugin API that expects plugins to do this, but in that case we should search only the main exe so we don't get collisions making things inconsistent.

Reply:
Thoughts was more that its a low level method in utils lib and the old implementation (that didn't do the right thing) seemed to have the intent to run it through all paths of mono_dl_lookup_symbol even in this case of ordinal, but didn't work due to incorrect implementation.

Ordinal exports should normally keep their numbers, or it will break clients.

I agree that the fallback looking through all loaded modules is incorrect and not something you would do because you will have no idea what you will get back.

But since ordinals is an edge case apparently not working at all in the past, I'm fine with just looking at the current module.